### PR TITLE
ProfileBundles: Better ImageStream support and more tests

### PR DESCRIPTION
**Pull request by JAORMX**
This adds better support for image streams by adding better validations when
parsing the reference. As part of this, it was easy to add support for image
streams in other namespaces, so that was done.

Finally, I added test cases for invalid image references to our e2e coverage.

### CODE REVIEW

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now ccb6556e004f39ce3a97ec30dfe1207caad16ada

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now 7bc837a39ad12ca2bbf63071f90e2dfacce6f04b

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now 4ab3eec9f28b49e9ab8e8c5e6a1b671cd95e4c18

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now dedb24d8cd861bcd7133478407909454a61b448a

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now e20d9f0a31890ee505062af6c89144156e45b052

File: pkg/controller/profilebundle/profilebundle_controller.go [hunk 1]
```diff
@@ -120,12 +121,27 @@ func (r *ReconcileProfileBundle) Reconcile(request reconcile.Request) (reconcile
...
+		pbCopy := instance.DeepCopy()
+		pbCopy.Status.DataStreamStatus = compliancev1alpha1.DataStreamInvalid
+		pbCopy.Status.ErrorMessage = err.Error()
+		err = r.client.Status().Update(context.TODO(), pbCopy)
+		if err != nil {
+			reqLogger.Error(err, "Couldn't update ProfileBundle status")
+			return reconcile.Result{}, err
+		}
+		// this was a fatal error, don't requeue
+		return reconcile.Result{}, nil
```
Comment (mrogers950) [inline pkg/controller/profilebundle/profilebundle_controller.go:137]: If there was a conflict with the Update() why do we not want to try again?

Comment (JAORMX) [reply pkg/controller/profilebundle/profilebundle_controller.go:137]: oh, this comment is not applicable. Probably copy-pasted it from elsewhere. If there's a conflict it'll retry, which is done in the if statement above.

Comment (mrogers950) [reply pkg/controller/profilebundle/profilebundle_controller.go:137]: ah yes, misread

**Pull request merged**